dags/aps/parser.py

```diff
@@ -0,0 +1,108 @@
+import re
+
+from aps.utils import get_nested
+from common.parsing.json_extractors import CustomExtractor, ValueExtractor
+from common.parsing.parser import IParser
+from structlog import get_logger
+
+
+class APSParser(IParser):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+        article_type_mapping = {
+            "article": "article",
+            "erratum": "erratum",
+            "editorial": "editorial",
+            "retraction": "retraction",
+            "essay": "other",
+            "comment": "other",
+            "letter-to-editor": "other",
+            "rapid": "other",
+            "brief": "other",
+            "reply": "other",
+            "announcement": "other",
+            "nobel": "other",
+        }
+
+        extractors = [
+            ValueExtractor("dois", keys=("identifiers", "doi")),
+            ValueExtractor(
+                "journal_doctype",
+                keys=("articleType",),
```

Comment:
😅 indeed, tuples are more memory efficient, but in our case, it doesn't make any difference because we don't have big arrays. Hence, it doesn't affect the result.

Also, just for reference, the main difference between a list and a tuple is that list is mutable, and a tuple is immutable. More details [here](https://www.geeksforgeeks.org/python-difference-between-list-and-tuple/)